File: 프로그래머스/0/181917. 간단한 논리 연산/간단한 논리 연산.cpp
#include <string>
#include <vector>

using namespace std;

bool func_or(bool x1, bool x2) {
    if (x1 == true && x2 == true)
        return true;
    else if ((x1 == true && x2 == false) || (x1 == false && x2 == true))
        return true;
    else if (x1 == false && x2 == false)
        return false;
}


bool func_and(bool x1, bool x2) {
    if (x1 == true && x2 == true)
        return true;
    else if ((x1 == true && x2 == false) || (x1 == false && x2 == true))
        return false;
    else if (x1 == false && x2 == false)
        return false;
}


bool solution(bool x1, bool x2, bool x3, bool x4) {
    bool answer = true;
    
    answer = func_and(func_or(x1, x2), func_or(x3, x4));
    
    return answer;
}

File: 프로그래머스/0/181917. 간단한 논리 연산/간단한 논리 연산.py
def func_or( x1,  x2) :
    if (x1 == True and x2 == True):
        return True
    elif ((x1 == True and x2 == False) or (x1 == False and x2 == True)):
        return True
    elif (x1 == False and x2 == False):
        return False



def func_and( x1,  x2) :
    if (x1 == True and x2 == True):
        return True
    elif ((x1 == True and x2 == False) or (x1 == False and x2 == True)):
        return False
    elif (x1 == False and x2 == False):
        return False



def solution(x1, x2, x3, x4):
    answer = True
    
    answer = func_and(func_or(x1, x2), func_or(x3, x4))
    return answer